src/traces/parcoords/parcoords.js

```diff
@@ -0,0 +1,762 @@
+        .attr('text-anchor', 'middle')
+        .style('font-weight', 100)
+        .style('font-size', '10px')
+        .style('cursor', 'default')
+        .style('user-select', 'none');
+}
+
+module.exports = function(root, svg, styledData, layout, callbacks) {
+
+    var domainBrushing = false;
+    var linePickActive = true;
+
+    function enterSvgDefs(root) {
+        var defs = root.selectAll('defs')
+            .data(repeat, keyFun);
+
+        defs.enter()
+            .append('defs');
+
+        var filterBarPattern = defs.selectAll('#filterBarPattern')
+            .data(repeat, keyFun);
+
+        filterBarPattern.enter()
+            .append('pattern')
+            .attr('id', 'filterBarPattern')
+            .attr('patternUnits', 'userSpaceOnUse');
+
+        filterBarPattern
+            .attr('x', -c.bar.width)
+            .attr('width', c.bar.capturewidth)
+            .attr('height', function(d) {return d.model.height;});
+
+        var filterBarPatternGlyph = filterBarPattern.selectAll('rect')
+            .data(repeat, keyFun);
+
+        filterBarPatternGlyph.enter()
+            .append('rect')
+            .attr('shape-rendering', 'crispEdges');
+
+        filterBarPatternGlyph
+            .attr('height', function(d) {return d.model.height;})
+            .attr('width', c.bar.width)
+            .attr('x', c.bar.width / 2)
+            .attr('fill', c.bar.fillcolor)
+            .attr('fill-opacity', c.bar.fillopacity)
+            .attr('stroke', c.bar.strokecolor)
+            .attr('stroke-opacity', c.bar.strokeopacity)
+            .attr('stroke-width', c.bar.strokewidth);
+    }
+
+    var vm = styledData
+        .filter(function(d) {return !!d.dimensions && d.dimensions.length > 0;})
+        .map(model.bind(0, layout))
+        .map(viewModel);
+
+    var parcoordsLineLayers = root.selectAll('.parcoords-line-layers')
+        .data(vm, keyFun);
+
+    parcoordsLineLayers.exit().remove();
+
+    parcoordsLineLayers.enter()
+        .insert('div', '.' + svg.attr('class').split(' ').join(' .')) // not hardcoding .main-svg
+        .classed('parcoords-line-layers', true)
+        .style('box-sizing', 'content-box');
+
+    parcoordsLineLayers
+        .style('transform', function(d, i) {
+            var translateY = i * (d.model.height + d.model.pad.t + d.model.pad.b);
+            return 'translate(' + d.model.translateX + 'px,' + (d.model.translateY + translateY) + 'px)';
+        });
+
+    var parcoordsLineLayer = parcoordsLineLayers.selectAll('.parcoords-lines')
+        .data(lineLayerModel, keyFun);
+
+    var tweakables = {renderers: [], dimensions: []};
+
+    var lastHovered = null;
+
+    parcoordsLineLayer.enter()
+        .append('canvas')
+        .attr('class', function(d) {return 'parcoords-lines ' + (d.context ? 'context' : d.pick ? 'pick' : 'focus');})
+        .style('box-sizing', 'content-box')
+        .style('transform', 'translate(' + (-c.overdrag) + 'px, 0)')
```

Comment:
This here is incompatible with the current `master` since commit https://github.com/plotly/plotly.js/pull/1332/commits/4aa1a4535a63633f38256159fc751ca59dd7b2bd where @alexcjohnson made the test dashboard catch negative `translate` values which can cause issue in some browsers (e.g. I.E.)

Trying to plot any of the `gl2d_parcoords_*` mock in the test dashboards leads to:

![image](https://cloud.githubusercontent.com/assets/6675409/22796670/37a234f6-eec9-11e6-8cb6-429870fed714.png)